### ISSUE TITLE
Validation error from defined query after upgrade of network model

### ISSUE THREAD

**Issue by jdockter** [hyperledger/composer#4189]
Using Composer 0.19.9 and IBP Starter in Prod...deployed my network, upgraded several times, no problem. Then I updated a field on an asset in my Composer model (minor change to put one letter to lower case)...I also use that same field in a defined query in my queries.qry file. I restarted my Composer REST server to pickup the new model changes, I ran several node-red transactions to create the asset I changed no problem, however when I go to run a query I get an validation error on that field I changed. I know Composer will generate a couchdb index, wondering if something isn't getting refreshed and some how returning the field with the wrong name, thus the validation error I'm seeing? Is an old asset in the world state getting returned in the query and causing a validation error?

When running local and on our dev server this works fine, although that is on a fresh install of the new model.
## Expected Behavior
Should be able to run query without validation error because model has been changed.

## Actual Behavior

Error back from REST server connected to Starter Plan when running any query (correct field name is masterAirwayBillNumber) :
{
"error": {
"statusCode": 500,
"name": "Error",
"message": "2 UNKNOWN: error executing chaincode: transaction returned with failure: ValidationException: Unexpected properties for type com.ibm.customs.Shipment: masterAirWayBillNumber",
"code": 2,
"metadata": {
"_internal_repr": {
"x-envoy-upstream-service-time": [
"43"
],
"date": [
"Tue, 26 Jun 2018 02:24:53 GMT"
],
"server": [
"ibmblockchain"
]
}
},
"details": "error executing chaincode: transaction returned with failure: ValidationException: Unexpected properties for type com.ibm.customs.Shipment: masterAirWayBillNumber",
"stack": "Error: 2 UNKNOWN: error executing chaincode: transaction returned with failure: ValidationException: Unexpected properties for type com.ibm.customs.Shipment: masterAirWayBillNumber\n at new createStatusError (/home/composer/.npm-global/lib/node_modules/composer-rest-server/node_modules/grpc/src/client.js:64:15)\n at /home/composer/.npm-global/lib/node_modules/composer-rest-server/node_modules/grpc/src/client.js:583:15"
}
}

## Steps to Reproduce

1. Create model with Asset field 'abc' as optional
2. Create defined query using 'abc' or just the Asset 'abc' is apart of
3. Deploy to network
4. Create Asset in world state with 'abc'
5. Update field 'abc' to 'abcDef' (or whatever value)...must be the same field, remain optional
6. Upgrade the network with new model
7. Run query, should get validation error about 'abc' being unexpected property

## Existing issues

- [ ] [Stack Overflow issues](http://stackoverflow.com/tags/hyperledger-composer)
- [ ] [GitHub Issues](https://github.com/hyperledger/composer/issues)
- [ ] [Rocket Chat history](https://chat.hyperledger.org/channel/composer)

## Your Environment

* Version used:
* Environment name and version (e.g. Chrome 39, node.js 5.4):
* Operating System and version (desktop or mobile):
* Link to your project:

**Comment by jdockter** [comment]
So added back in the field I changed, ie 'abc', along with 'abcDef' and redeployed to network.  Was then able to see the Asset again via REST server with 'abc'.  Deleted the Assets with 'abc'.  Changed model again back to only 'abcDef' and did a network upgrade.  Now I run query again via REST server.  So something with the old data was causing the validation error.  Not sure if that is normal or not.  @andrew-coleman see above
